pkg/trace/writer/trace.go

```diff
@@ -98,26 +103,22 @@ func NewTraceWriter(cfg *config.AgentConfig, prioritySampler samplerTPSReader, e
 	}
 	climit := cfg.TraceWriter.ConnectionLimit
 	if climit == 0 {
-		// Default to 10% of the connection limit to outgoing sends.
-		// Since the connection limit was removed, keep this at 200
-		// as it was when we had it (2k).
-		climit = 200
+		climit = 100
```

Comment:
Where did this 100 come from?

Reply:
This was arrived at through experimentation. We can leave it at 200 if we want and I can retest.

I wanted to bring this down, the argument being that submitting payloads is inherently network-bound, not CPU bound, so the benefits of additional connections falls off precipitously, resulting only in additional RAM usage. The bandwidth is only so high and throwing concurrency at it will not help. 

However, I didn't want to be too aggressive. Cutting the former limit in half seemed like a good first step and with this configuration I got good results.